mesonbuild/modules/cmake.py

```diff
@@ -136,6 +136,8 @@ def dependency(self, state: ModuleState, args: T.List[str], kwargs: T.Dict[str,
         info = self._args_to_info(args)
         if info['func'] == 'executable':
             raise InvalidArguments(f'{args[0]} is an executable and does not support the dependency() method. Use target() instead.')
+        if info['dep'] is None:
+            raise InvalidArguments(f'{args[0]} does not support the dependency() method. Use target() instead.')
```

Comment:
Afaik, there is no `target()` method in meson...

Reply:
The `dependency()` referred is not the `meson` dependency either - it is the `cmake` one.....

Reply:
Oh! Sorry, I didn't realised that... You are right.